src/vs/workbench/parts/extensions/browser/extensionsList.ts

```diff
@@ -149,6 +149,20 @@ export class Renderer implements IPagedRenderer<IExtension, ITemplateData> {
 			toggleClass(data.root, 'disabled', isInstalled && !isExtensionRunning);
 		});
 
+
+		data.extensionDisposables.push(this.extensionsWorkbenchService.onRecommendationChange(() => {
+			const extRecommendations = this.extensionTipsService.getAllRecommendationsWithReason();
+			if (!extRecommendations[extension.id.toLowerCase()]) {
+				data.root.setAttribute('aria-label', extension.displayName);
+				data.root.title = '';
+				removeClass(data.root, 'recommended');
+			} else {
+				data.root.setAttribute('aria-label', extension.displayName + '. ' + extRecommendations[extension.id]);
+				data.root.title = extRecommendations[extension.id.toLowerCase()].reasonText;
+				addClass(data.root, 'recommended');
+			}
```

Comment:
The logic above is repeated a few lines below. I would suggest to move this logic into a function `updateRecommendation` and use it in both places.

Also, let's move creating this disposable to after setting the recommendation related ui below. Both are related and it helps readability if they are next to each other.